types/underscore/underscore-tests.ts

```diff
@@ -209,6 +209,28 @@ _(stooges)
 	.indexBy('age')
 	.value()['40'].age;
 
+let pensioners: string[] = _.chain(stooges)
+    .filter(p => p.age >= 60)
+    .map(p => p.name)
+    .value();
+
+var usersData: _.Dictionary<{ age: number; name: string }> = {
+    'user id': { name: 'moe', age: 40 },
+    'other user Id': { name: 'larry', age: 50 },
+    'fake id': { name: 'curly', age: 60 },
+};
+
+let youngPeopleId: string[] = _.chain(usersData)
```

Comment:
This test should actually fail, as performing filter on a a Dictionary<T> returns T[], thus performing map on it will need the following iterator:
```
(value: T, index: number, list: T[]): TResult
```
The reason this test passes now is because of missing signature of filter for _Chain which should be:
```ts
filter(iterator: _.ObjectIterator<T, boolean>, context?: any): _Chain<T, T[]>;
```
But fixing it will not even help as the entire _Chain interface is malformed :-( and I'm working on a wide fix for it.
I would rather you only add tests that use the fixed map method

Reply:
You have right. I will stick in this test to 'map' function. I interface definition I was trying to test.
And, I will add some test for 'filter' separate.